// slash-command-builder/build/options/SlashCommandChannelOption.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const lodash_1 = require("lodash");
const interfaces_1 = require("../interfaces");
const SlashCommandOption_1 = __importDefault(require("../commons/SlashCommandOption"));
const allowTypes = [
    interfaces_1.ChannelTypes.GUILD_TEXT,
    interfaces_1.ChannelTypes.DM,
    interfaces_1.ChannelTypes.GUILD_VOICE,
    interfaces_1.ChannelTypes.GROUP_DM,
    interfaces_1.ChannelTypes.GUILD_CATEGORY,
    interfaces_1.ChannelTypes.GUILD_NEWS,
    interfaces_1.ChannelTypes.GUILD_STORE,
    interfaces_1.ChannelTypes.GUILD_NEWS_THREAD,
    interfaces_1.ChannelTypes.GUILD_PUBLIC_THREAD,
    interfaces_1.ChannelTypes.GUILD_PRIVATE_THREAD,
    interfaces_1.ChannelTypes.GUILD_STAGE_VOICE,
];
class SlashCommandChannelOption extends SlashCommandOption_1.default {
    constructor() {
        super(interfaces_1.ApplicationCommandOptionTypes.Channel);
        this.type = interfaces_1.ApplicationCommandOptionTypes.Channel;
        this.channelTypes = [];
    }
    addFilterBy(value) {
        let deny = true;
        if ((0, lodash_1.isArray)(value)) {
            deny = (0, lodash_1.isEmpty)(value);
        }
        else {
            deny = !(0, lodash_1.isNumber)(value);
        }
        if (deny) {
            throw new Error('The `value` parameter is not an array/ChannelType or is empty');
        }
        if ((0, lodash_1.isArray)(value)) {
            for (const item of value) {
                if (!this.channelTypes.includes(item)) {
                    if (!this.checkType(item)) {
                        break;
                    }
                    this.channelTypes.push(item);
                }
            }
        }
        else if (!this.channelTypes.includes(value)) {
            if (!this.checkType(value)) {
                return this;
            }
            this.channelTypes.push(value);
        }
        return this;
    }
    toJSON() {
        return {
            ...super.toJSON(),
            channel_types: this.channelTypes,
        };
    }
    checkType(type) {
        let validate = allowTypes.includes(type);
        if (!validate) {
            throw new Error('The `value` parameter is not a ChannelType');
        }
        return validate;
    }
}
exports.default = SlashCommandChannelOption;
