// slash-command-builder/build/commons/SlashCommandMethods.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const lodash_1 = require("lodash");
const SlashCommandBooleanOption_1 = __importDefault(require("../options/SlashCommandBooleanOption"));
const SlashCommandChannelOption_1 = __importDefault(require("../options/SlashCommandChannelOption"));
const SlashCommandIntegerOption_1 = __importDefault(require("../options/SlashCommandIntegerOption"));
const SlashCommandMentionableOption_1 = __importDefault(require("../options/SlashCommandMentionableOption"));
const SlashCommandNumberOption_1 = __importDefault(require("../options/SlashCommandNumberOption"));
const SlashCommandRoleOption_1 = __importDefault(require("../options/SlashCommandRoleOption"));
const SlashCommandStringOption_1 = __importDefault(require("../options/SlashCommandStringOption"));
const SlashCommandUserOption_1 = __importDefault(require("../options/SlashCommandUserOption"));
class SlashCommandMethods {
    constructor() {
        this.options = [];
    }
    addBooleanOption(callback) {
        this.addNewOption(callback, SlashCommandBooleanOption_1.default);
        return this;
    }
    addChannelOption(callback) {
        this.addNewOption(callback, SlashCommandChannelOption_1.default);
        return this;
    }
    addIntegerOption(callback) {
        this.addNewOption(callback, SlashCommandIntegerOption_1.default);
        return this;
    }
    addMentionableOption(callback) {
        this.addNewOption(callback, SlashCommandMentionableOption_1.default);
        return this;
    }
    addNumberOption(callback) {
        this.addNewOption(callback, SlashCommandNumberOption_1.default);
        return this;
    }
    addRoleOption(callback) {
        this.addNewOption(callback, SlashCommandRoleOption_1.default);
        return this;
    }
    addStringOption(callback) {
        this.addNewOption(callback, SlashCommandStringOption_1.default);
        return this;
    }
    addUserOption(callback) {
        this.addNewOption(callback, SlashCommandUserOption_1.default);
        return this;
    }
    addNewOption(callback, Instance) {
        if (!(0, lodash_1.isFunction)(callback)) {
            throw new Error('The `callback` parameter is not a function');
        }
        this.options.push(callback(new Instance()).toJSON());
        return this;
    }
}
exports.default = SlashCommandMethods;
