// slash-command-builder/build/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.SlashCommandUserOption = exports.SlashCommandStringOption = exports.SlashCommandRoleOption = exports.SlashCommandNumberOption = exports.SlashCommandMentionableOption = exports.SlashCommandIntegerOption = exports.SlashCommandChannelOption = exports.SlashCommandBooleanOption = exports.SlashCommandSubcommandGroup = exports.SlashCommandSubcommand = exports.ChannelTypes = exports.ApplicationCommandTypes = exports.ApplicationCommandOptionTypes = void 0;
require('module-alias/register');
const SlashCommandBooleanOption_1 = __importDefault(require("./options/SlashCommandBooleanOption"));
exports.SlashCommandBooleanOption = SlashCommandBooleanOption_1.default;
const SlashCommandChannelOption_1 = __importDefault(require("./options/SlashCommandChannelOption"));
exports.SlashCommandChannelOption = SlashCommandChannelOption_1.default;
const SlashCommandIntegerOption_1 = __importDefault(require("./options/SlashCommandIntegerOption"));
exports.SlashCommandIntegerOption = SlashCommandIntegerOption_1.default;
const SlashCommandMentionableOption_1 = __importDefault(require("./options/SlashCommandMentionableOption"));
exports.SlashCommandMentionableOption = SlashCommandMentionableOption_1.default;
const SlashCommandNumberOption_1 = __importDefault(require("./options/SlashCommandNumberOption"));
exports.SlashCommandNumberOption = SlashCommandNumberOption_1.default;
const SlashCommandRoleOption_1 = __importDefault(require("./options/SlashCommandRoleOption"));
exports.SlashCommandRoleOption = SlashCommandRoleOption_1.default;
const SlashCommandStringOption_1 = __importDefault(require("./options/SlashCommandStringOption"));
exports.SlashCommandStringOption = SlashCommandStringOption_1.default;
const SlashCommandUserOption_1 = __importDefault(require("./options/SlashCommandUserOption"));
exports.SlashCommandUserOption = SlashCommandUserOption_1.default;
var interfaces_1 = require("./interfaces");
Object.defineProperty(exports, "ApplicationCommandOptionTypes", { enumerable: true, get: function () { return interfaces_1.ApplicationCommandOptionTypes; } });
Object.defineProperty(exports, "ApplicationCommandTypes", { enumerable: true, get: function () { return interfaces_1.ApplicationCommandTypes; } });
Object.defineProperty(exports, "ChannelTypes", { enumerable: true, get: function () { return interfaces_1.ChannelTypes; } });
const main_1 = require("./main");
exports.default = main_1.SlashCommandBuilder;
var main_2 = require("./main");
Object.defineProperty(exports, "SlashCommandSubcommand", { enumerable: true, get: function () { return main_2.SlashCommandSubcommand; } });
Object.defineProperty(exports, "SlashCommandSubcommandGroup", { enumerable: true, get: function () { return main_2.SlashCommandSubcommandGroup; } });
